pci/src/configuration.rs

```diff
@@ -432,6 +448,37 @@ impl PciConfiguration {
         }
     }
 
+    fn state(&self) -> PciConfigurationState {
+        PciConfigurationState {
+            registers: self.registers.to_vec(),
+            writable_bits: self.writable_bits.to_vec(),
+            bar_addr: self.bar_addr.to_vec(),
+            bar_size: self.bar_size.to_vec(),
+            bar_used: self.bar_used.to_vec(),
+            bar_type: self.bar_type.to_vec(),
```

Comment:
Not to do with your change, but it highlights it. Maybe we should store a `Vec<PciBar>` where `struct PciBar` contains the address size, type, etc.

Reply:
If we can make it work, that would be cleaner indeed. Let me open a separate issue for this.

Reply:
https://github.com/cloud-hypervisor/cloud-hypervisor/issues/1153